graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/context/DgsContext.kt

```diff
@@ -60,4 +61,13 @@ open class DgsContext(val customContext: Any? = null, val requestData: DgsReques
             return dgsContext.requestData
         }
     }
+
+    /**
+     * This allows optimization for consumers who use FederatedTracingInstrumentation
+     * by providing it access to the incoming headers to determine if it can short-circuit
+     * the tracing if the request does not originate from a Gateway.
+     */
+    override fun getHTTPRequestHeader(caseInsensitiveHeaderName: String): String? {
```

Comment:
Making the request context implement the Apollo federated interface makes me raise an eyebrow as its Apollo specific and it may not always be applicable. Maybe just mention this and see if the DGS team has concerns or suggestions on making it "cleaner/purer" or whatever the term I am looking for is. :)

Reply:
Yeah.  I started experimenting with another potential solution for this -- adding more flexibility into the federation-jvm tracing class to no longer require clients to implement a particular interface, which would allow codebases using both DGS and federation-jvm to provide the glue if we can't come up with a clean and simple way to have this wired in automatically by DGS. 

Have a look: https://github.com/apollographql/federation-jvm/pull/116